pkg/util/paramtable/grpc_param.go

```diff
@@ -130,15 +142,52 @@ func (p *grpcConfig) init(domain string, base *BaseTable) {
 		Export:  true,
 	}
 	p.CaPemPath.Init(base.mgr)
+
+	p.InternalTLSEnabled = ParamItem{
+		Key:          "common.security.internaltlsEnabled",
+		Version:      "2.0.0",
+		DefaultValue: "0",
+		Export:       true,
+	}
+	p.InternalTLSEnabled.Init(base.mgr)
+
+	p.InternalTLSServerPemPath = ParamItem{
+		Key:     "internaltls.serverPemPath",
+		Version: "2.0.0",
+		Export:  true,
+	}
+	p.InternalTLSServerPemPath.Init(base.mgr)
+
+	p.InternalTLSServerKeyPath = ParamItem{
+		Key:     "internaltls.serverKeyPath",
+		Version: "2.0.0",
+		Export:  true,
+	}
+	p.InternalTLSServerKeyPath.Init(base.mgr)
+
+	p.InternalTLSCaPemPath = ParamItem{
+		Key:     "internaltls.caPemPath",
+		Version: "2.0.0",
+		Export:  true,
+	}
+	p.InternalTLSCaPemPath.Init(base.mgr)
 }
 
 // GetAddress return grpc address
 func (p *grpcConfig) GetAddress() string {
-	return p.IP + ":" + p.Port.GetValue()
+	if !p.InternalTLSEnabled.GetAsBool() {
+		return p.IP + ":" + p.Port.GetValue()
+	}
+	fmt.Println("address: ", p.Address.GetValue())
+	return p.Address.GetValue() + ":" + p.Port.GetValue()
 }
 
 func (p *grpcConfig) GetInternalAddress() string {
-	return p.IP + ":" + p.InternalPort.GetValue()
+	if !p.InternalTLSEnabled.GetAsBool() {
+		return p.IP + ":" + p.InternalPort.GetValue()
+	}
+	fmt.Println("address: ", p.Address.GetValue())
+	return p.Address.GetValue() + ":" + p.InternalPort.GetValue()
```

Comment:
Do we need separate configurations for `p.IP` and `p.Address`? Can we reuse the existing configuration `p.IP`, with the understanding that you can use a hostname instead of the IP address if desired?

Reply:
That's what I'm wondering.
what if we simply use ip instead?

Reply:
It works, I have modified the code.
It will pick from the IP configuration from the milvus.yaml